### PR TITLE
Document API Tokens and Scopes

### LINKED ISSUE #2325: document API scopes
The UI gives the option to create an API token and specify the scope, but doesn't explain what the different scopes mean.

screen shot 2018-05-11 at 10 25 57 am

Specific questions:

* Which scopes gives read access, vs. read-write?
* Does `Status` give any permissions beyond being able to see the status badge?

### PR DESCRIPTION
**Pull request by smart-alek**
This PR creates a new document that outlines the different API token types, as well as how to create/delete them. The document also links to some other documents which require API access.

@ndintenfass for technical accuracy
@michelle-luna for language edits

Firmly resolves #2325 and [this JIRA issue](https://circleci.atlassian.net/browse/CIRCLE-11262).

### CODE REVIEW

File: jekyll/_cci2/api-permissions.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+To use the CircleCI API
+or view details about a build,
+you will need API tokens with the appropriate permissions.
+This document describes the types of API tokens,
+as well as how to create and delete them.
+
+## Overview
+
+There are two types of API token
```
Comment (michelle-luna) [inline jekyll/_cci2/api-permissions.md]: token-> tokens

File: jekyll/_cci2/api-permissions.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+  These tokens provide information about specific projects
+  and only grant read permissions.
+  Project tokens have three scope options: _Status_, _Build Artifacts_, and _All_.
+      - _Status_ tokens grant read access to the project's build statuses.
+      Useful for [embedding status badges]({{ site.baseurl }}/2.0/status-badges/).
+      - _Build Artifacts_ tokens grant read access to the project's build artifacts.
+      - _All_ tokens grant read access for the project's build statuses and build artifacts.
+
+**Note**:
+Once created,
```
Comment (michelle-luna) [inline jekyll/_cci2/api-permissions.md]: Once-> After you create an API token (once is colloquial and could be confused for 'one time'

File: jekyll/_cci2/api-permissions.md [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+is to delete and recreate them.
+
+### Creating a Personal API Token
+
+  1. In the CircleCI application,
+  go to your [User settings](https://circleci.com/account).
+  2. Click [Personal API Tokens](https://circleci.com/account/api).
+  3. Click the **Create New Token** button.
+  4. In the **Token name** field,
+  create a memorable name for the token.
```
Comment (michelle-luna) [inline jekyll/_cci2/api-permissions.md]: create -> type

File: jekyll/_cci2/api-permissions.md [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+
+  1. In the CircleCI application,
+  go to your project's settings
+  by clicking the gear icon next to your project.
+  2. In the **Permissions** section,
+  click on **API Permissions**.
+  3. Click the **Create Token** button.
+  4. Choose a scope from the dropdown menu.
+  5. In the **Token Label** field,
+  create a memorable label for the token.
```
Comment (michelle-luna) [inline jekyll/_cci2/api-permissions.md]: create-> type

File: jekyll/_cci2/api-permissions.md [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+  6. Click the **Add Token** button.
+
+To delete a project API token,
+click the X in the **Remove** column.
+When the confirmation window appears,
+click the **Remove** button.
+
+## What Next?
```
Comment (michelle-luna) [inline jekyll/_cci2/api-permissions.md]: What Next-> Next Steps (headings should not be questions unless in an FAQ)

File: jekyll/_cci2/api-permissions.md [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+
+Now that you know
+how to create API tokens,
+you could...
+
+  - translate your legacy project to CircleCI 2.0 with the [configuration translator]({{ site.baseurl }}/2.0/config-translation/#using-config-translation).
```
Comment (michelle-luna) [inline jekyll/_cci2/api-permissions.md]: Translate use capital to start sentence

File: jekyll/_cci2/api-permissions.md [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+  - [inject environment variables]({{ site.baseurl }}/2.0/env-vars/#injecting-environment-variables-with-the-api) into a build.
```
Comment (michelle-luna) [inline jekyll/_cci2/api-permissions.md]: Inject

File: jekyll/_cci2/api-permissions.md [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+  - [trigger conditional jobs]({{ site.baseurl }}/2.0/api-job-trigger/).
```
Comment (michelle-luna) [inline jekyll/_cci2/api-permissions.md]: Trigger

File: jekyll/_cci2/api-permissions.md [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+  - learn how to [embed build status badges]({{ site.baseurl }}/2.0/status-badges/) in your project's README.
```
Comment (michelle-luna) [inline jekyll/_cci2/api-permissions.md]: Learn
Also use initial caps for the document title

File: jekyll/_cci2/api-permissions.md [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+  - [download a build's artifacts]({{ site.baseurl }}/2.0/artifacts/#downloading-all-artifacts-for-a-build-on-circleci) for safekeeping.
```
Comment (michelle-luna) [inline jekyll/_cci2/api-permissions.md]: Download

File: jekyll/_cci2/api-permissions.md [hunk 11]
```diff
@@ -0,0 +1,82 @@
+---
+layout: classic-docs
+title: "API Permissions"
```
Comment (michelle-luna) [inline jekyll/_cci2/api-permissions.md]: Using API Permissions? Setting API Permissions? Title should start with action verb.

File: jekyll/_cci2/api-permissions.md [hunk 12]
```diff
@@ -0,0 +1,82 @@
...
+you will need API tokens with the appropriate permissions.
+This document describes the types of API tokens,
+as well as how to create and delete them.
+
+## Overview
+
+There are two types of API token
+you can create within CircleCI.
+
+  1. **Personal**:
```
Comment (michelle-luna) [inline jekyll/_cci2/api-permissions.md]: 1. -> - 
Generally, only use numbers for steps, lists are bullets

File: jekyll/_cci2/api-permissions.md [hunk 13]
```diff
@@ -0,0 +1,82 @@
...
+  These tokens are used to interact with the CircleCI API
+  and grant full read and write permissions.
+  2. **Project**:
+  These tokens provide information about specific projects
+  and only grant read permissions.
```
Comment (ndintenfass) [inline jekyll/_cci2/api-permissions.md]: You can't use Project tokens to trigger?

File: jekyll/_cci2/api-permissions.md [hunk 14]
```diff
@@ -0,0 +1,82 @@
...
+click the X in the **Remove** column.
+When the confirmation window appears,
+click the **Remove** button.
+
+## What Next?
+
+Now that you know
+how to create API tokens,
+you could...
+
```
Comment (ndintenfass) [inline jekyll/_cci2/api-permissions.md]: ^^ this tone feels different than our usual - is that intended? (curious also why you are writing in such short lines - poetry?)

File: jekyll/_cci2/api-permissions.md (hunk 6 above)
Comment (ndintenfass) [reply jekyll/_cci2/api-permissions.md]: Surprised to see this as the first thing - you can run the translator from your browser using the cookie, which i thought was the main people people did it?

File: jekyll/_cci2/api-permissions.md (hunk 2 above)
Comment (smart-alek) [reply jekyll/_cci2/api-permissions.md]: [Added to Style Guide](https://github.com/circleci/circleci-docs/wiki/Style-Guide#once-upon-a-time)

File: jekyll/_cci2/api-permissions.md (hunk 11 above)
Comment (smart-alek) [reply jekyll/_cci2/api-permissions.md]: I think "Using" applies more to the "Next Steps" section. "Setting" seems fine, but what do you think about "Managing"?

File: jekyll/_cci2/basics.md [hunk 15]
```diff
@@ -29,6 +29,7 @@ Document | Description
 [Using Shell Scripts]({{ site.baseurl }}/2.0/using-shell-scripts/) | Best practices for using shell scripts in CircleCI configuration.
 [Adding an SSH Key]({{ site.baseurl }}/2.0/add-ssh-key/) | How to add an SSH key to CircleCI.
 [Debugging with SSH]({{ site.baseurl }}/2.0/ssh-access-jobs/) | How to use SSH to debug build problems.
+[API Permissions]({{ site.baseurl }}/2.0/api-permissions/) | How to add and change API permissions on CircleCI.
```
[the remark this answers is not in the document]
Comment (smart-alek) [reply jekyll/_cci2/basics.md]: Nice — was feeling iffy about that title already.

File: jekyll/_cci2/api-permissions.md [hunk 16]
```diff
@@ -0,0 +1,82 @@
...
+To delete a project API token,
+click the X in the **Remove** column.
+When the confirmation window appears,
+click the **Remove** button.
+
+## What Next?
+
+Now that you know
+how to create API tokens,
+you could...
```
[the remark this answers is not in the document]
Comment (smart-alek) [reply jekyll/_cci2/api-permissions.md]: Good question. I was thinking that that sort of thing would be better served by a page devoted to **using the API**. But I see the pros of pushing a link on how to do that to the top of this list.

File: jekyll/_cci2/api-permissions.md (hunk 14 above)
Comment (smart-alek) [reply jekyll/_cci2/api-permissions.md]: Ha. It is probably a bit too casual. The line spacing is something called [semantic linefeeds](http://rhodesmill.org/brandon/2012/one-sentence-per-line/). I've been using it to make editing and review easier.

File: jekyll/_cci2/api-permissions.md (hunk 13 above)
Comment (smart-alek) [reply jekyll/_cci2/api-permissions.md]: Hmm. If that's the case, which scope would it need to be set at? The doc on [triggering jobs conditionally](https://circleci.com/docs/2.0/api-job-trigger/#conditionally-running-jobs-with-the-api) only says you need the "API token from your project's settings page".

**Pull request merged**